### ISSUE TITLE
The "Privacy Settings" doesn't appear when tapping the "Go to Settings" button of the Compliance Popover

### ISSUE THREAD

**Issue by salimbraksa** [wordpress-mobile/WordPress-iOS#21619]
### Expected behavior

1. User is presented with the Compliance Popover
2. User taps "Go to Settings"
3. The "Privacy Settings" should be presented in this order "Me > App Settings > Privacy Settings".

### Actual behavior

The "Me" is shown but the navigation to "Privacy Settings" doesn't happen. 

https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/6b78c652-9686-4460-b2a4-02567597fb42

### Steps to reproduce the behavior

1. Fresh install Jetpack app.
2. **Expect** the Compliance Popover to appear.
3. Tap "Go to Settings" button.
4. The "Me" screen is shown but the navigation to "Privacy Settings" is interrupted.

**Tip**

You can force display the Compliance Popover by making the following code changes:
1. Open `CompliancePopoverCoordinator.swift` file.
2. Force `shouldShowPrivacyBanner` method to return `true`.
3. Comment this line `!defaults.didShowCompliancePopup` in `presentIfNeeded` method.

**Comment by mokagio** [comment]
Looks like #21630 addressed this issue but didn't automatically close it, maybe because it landed on `release/23.3` instead of `trunk`?

Anyways, I'll updated the milestone to match and close this manually.
